### ISSUE TITLE
Can't Read Structure

### ISSUE THREAD

**Issue by WenXin20** [embeddedt/ModernFix#49]
```
Minecraft v1.19.2
Forge v43.2.7 
Modern Fix v3.2.1
```

After installing this mod on my server, loading new chunks causes the player to be kicked from the server and an error to print out in the log. It seems to have occurred with both ad astra and spelunkery mod structures, although the mods are probably not related to the issue.

<details>
  <summary> Click Here for Error Log</summary>
  
  ```
[27Mar2023 00:32:06.446] [Worker-Main-4/ERROR] [ModernFix/]: Can't read structure
java.io.FileNotFoundException: ad_astra:structures/venus_bullet.nbt
	at net.minecraft.server.packs.resources.ResourceProvider.m_215590_(ResourceProvider.java:20) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.server.packs.resources.ResourceProvider.m_215593_(ResourceProvider.java:20) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.server.packs.resources.ResourceProvider.m_215595_(ResourceProvider.java:24) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager.m_230427_(StructureTemplateManager.java:533) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager.m_230425_(StructureTemplateManager.java:102) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708) ~[?:?]
	at net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager.m_230407_(StructureTemplateManager.java:92) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager.m_230359_(StructureTemplateManager.java:80) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-5.0.28.jar%2375!/:?]
	at net.minecraft.world.level.levelgen.structure.pools.SinglePoolElement.m_227299_(SinglePoolElement.java:85) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.pools.SinglePoolElement.m_214015_(SinglePoolElement.java:118) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement.m_227238_(JigsawPlacement.java:295) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.structures.JigsawStructure.m_214086_(JigsawStructure.java:86) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.levelgen.structure.Structure.m_226596_(Structure.java:84) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.chunk.ChunkGenerator.m_223104_(ChunkGenerator.java:638) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.chunk.ChunkGenerator.m_223066_(ChunkGenerator.java:591) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at net.minecraft.world.level.chunk.ChunkGenerator.m_223164_(ChunkGenerator.java:573) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.chunk.ChunkStatus.m_223360_(ChunkStatus.java:59) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.world.level.chunk.ChunkStatus.m_223279_(ChunkStatus.java:292) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$25(ChunkMap.java:621) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-5.0.28.jar%2375!/:?]
	at net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$27(ChunkMap.java:619) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62) ~[server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.util.thread.ProcessorMailbox.m_18759_(ProcessorMailbox.java:91) [server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.util.thread.ProcessorMailbox.m_18747_(ProcessorMailbox.java:146) [server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at net.minecraft.util.thread.ProcessorMailbox.run(ProcessorMailbox.java:102) [server-1.19.2-20220805.130853-srg.jar%23603!/:?]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1176) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1647) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1614) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) [?:?]

  ```
  
</details>

**Comment by embeddedt** [comment]
You can add `mixin.perf.cache_upgraded_structures=false` to the `modernfix-mixins.properties` file to work around the issue for now. I will try to investigate when I have time.

**Comment by WenXin20** [comment]
Thanks!

**Comment by embeddedt** [comment]
Could you send the full log, please? I'm actually not sure if that is really the cause of the issue.

**Comment by P3rf3ctXZer0** [comment]
So the cause is because apparently Mojang actually removed sections of structures internally but didn't release an update framework. Repurposed structures dev is dealing with a similar issues with the data packs he created. Hashtag blame microjang XD

**Comment by embeddedt** [comment]
Log spam should be resolved as of 3.2.2. If the player continues to be kicked after upgrading to that version please open a new issue report.
